src/main/java/org/jetbrains/appenders/JsonLayout.java

```diff
@@ -105,15 +141,43 @@ public static Field fromValue(String val) {
         }
     }
 
+    private class RenderedFieldLabels {
+        String locationClass = getLocationFieldLabel("class");
+        String locationFile = getLocationFieldLabel("file");
+        String locationLine = getLocationFieldLabel("line");
+        String locationMethod = getLocationFieldLabel("method");
+
+        String exceptionClass = getExceptionFieldLabel("class");
+        String exceptionMessage = getExceptionFieldLabel("message");
+        String exceptionStacktrace = getExceptionFieldLabel("stacktrace");
+
+        String exception = getFieldLabel("exception");
+        String level = getFieldLabel("level");
+        String location = getFieldLabel("location");
+        String logger = getFieldLabel("logger");
+        String message = getFieldLabel("message");
+        String mdc = getFieldLabel("mdc");
+        String ndc = getFieldLabel("ndc");
+        String host = getFieldLabel("host");
+        String path = getFieldLabel("path");
+        String tags = getFieldLabel("tags");
+        String timestamp = getFieldLabel("@timestamp");
+        String thread = getFieldLabel("thread");
+        String version = getFieldLabel("@version");
+    }
+
     private static final String VERSION = "1";
 
     private String tagsVal;
     private String fieldsVal;
     private String includedFields;
     private String excludedFields;
+    private String[] renamedFieldLabels;
 
     private final Map<String, String> fields;
     private final Set<Field> renderedFields;
+    private RenderedFieldLabels renderedFieldLabels;
```

Comment:
you may have this variable as `private final RenderedFieldLabels renderedFieldLabels = new RenderedFieldLabels();` That makes the logger work even without initialization being called.

Reply:
Ok, I see how you implemented it. `final` will not work :)